prototypes/stack-sizes.lua

```diff
@@ -1,120 +1,46 @@
-local bzutil = require "__bzlead__/data-util";
-
--- These ones are already added by deadlock crating
-local raw_material_list = {
-  [1] = { "wood", "iron-ore", "copper-ore", "stone", "coal", "iron-plate", "copper-plate", "steel-plate", "stone-brick", "lead-ore", "lead-plate" },
-  [2] = { "copper-cable", "iron-gear-wheel", "iron-stick", "sulfur", "plastic-bar", "solid-fuel", "electronic-circuit", "advanced-circuit", "titanium-ore", "titanium-plate" },
-  [3] = { "processing-unit", "battery", "uranium-ore", "uranium-235", "uranium-238" },
-}
-
--- These are checked for existence before being added so it doesn't matter if their mod isn't loaded
-local new_material_list = {
-  [1] = { "sand", "glass", "stone-tablet" },  -- AAI Industry
-  [2] = {
-    "automation-science-pack", "logistic-science-pack", "ff-transport-science-pack", "chemical-science-pack", "military-science-pack",  -- Vanilla
-    "explosives", "engine-unit", "electric-engine-unit",  -- Vanilla
-    "ff-charged-battery",  -- Freight Forwarding
-    "motor", "electric-motor", "processed-fuel",  -- AAI Industry
-  },
-  [3] = {
-    "flying-robot-frame", "rocket-control-unit", "low-density-structure", "rocket-fuel", "nuclear-fuel",  -- Vanilla
-    "production-science-pack", "utility-science-pack", "space-science-pack",  -- Vanilla
-    "ff-battery-pack", "ff-charged-battery-pack", "ff-cobalt-ore", "ff-cobalt-concentrate", "ff-cobalt-ingot", "ff-titansteel-plate",  -- Freight Forwarding
-    "plutonium-238", "plutonium-239", "plutonium-fuel" },  -- Plutonium Energy
-}
-
-stack_size_override = {
+-- changes value of stacks
+function ff_restack(item_name)
+  local item = data.raw.item[item_name] or data.raw.tool[item_name]
+  if not item then return end
+  local stack_size = tonumber(item.stack_size)
+  if ff_stack_size_override[item_name] then item.stack_size = ff_stack_size_override[item_name]
+  elseif item.subgroup == "barrel"       then item.stack_size = 10
+  elseif item.subgroup == "raw-resource" then item.stack_size = 25
+  elseif item.subgroup == "raw-material" then item.stack_size = 50
+  elseif item.subgroup == "science-pack" then item.stack_size = 100
+  elseif stack_size >= 50 then item.stack_size = stack_size / 2 end
```

Comment:
I'm really not sure about this approach - why can't we just assume that the original stack sizes were correct? (and then halve them to fit in with the new balance).

Reply:
This is just a "quick" solution I came up with. Previous worked with vanilla but with mod compatibilities there's no consistency in stack sizes so only halving will produce 200->100  and 50->25, so by setting the value based off the subgroup instead, will give 200/50->25 for both.

It will still default to just `/2` for basically all intermediates but I thought going 25 for raw ores, 50 for plates, and 100 for science was already good for consistency across whole board, as you get the nice vanilla-ish stack progression/compression from ore to plates